src/Aws/General.hs

```diff
@@ -182,9 +183,10 @@ data Region
     | UsEast1
     | UsWest1
     | UsWest2
-    deriving (Show, Read, Eq, Ord, Enum, Bounded, Typeable)
+    | CustomEndpoint !T.Text
```

Comment:
I wonder if it would make sense to have a separate field for the port. Are there places in the code elsewhere where the port must be past separately?